core/domain/stats_services.py

```diff
@@ -201,3 +246,148 @@ def get_exploration_stats(exploration_id, exploration_version):
+        state_answers.exploration_id, state_answers.exploration_version,
+        state_answers.state_name, state_answers.interaction_id,
+        state_answers.answers_list)
+    state_answers_model.save()
+
+
+def get_state_answers(exploration_id, exploration_version, state_name):
+    """Get state answers domain object obtained from StateAnswersModel instance
+    stored in data store.
+    """
+    state_answers_model = stats_models.StateAnswersModel.get_model(
+        exploration_id, exploration_version, state_name)
+    if state_answers_model:
+        return stats_domain.StateAnswers(
+            exploration_id, exploration_version, state_name,
+            state_answers_model.interaction_id,
+            state_answers_model.answers_list)
+    else:
+        return None
+
+
+def _validate_answer(answer_dict):
+    """Validate answer dicts. In particular, check the following:
+
+    - Minimum set of keys: 'answer_value', 'time_spent_in_sec',
+            'session_id'
+    - Check size of every answer_value
+    - Check time_spent_in_sec is non-negative
+    """
+
+    # TODO(msl): These validation methods need tests to ensure that
+    # the right errors show up in the various cases.
+
+    # Minimum set of keys required for answer_dicts in answers_list
+    REQUIRED_ANSWER_DICT_KEYS = ['answer_value', 'time_spent_in_sec',
+                                 'session_id']
+
+    # There is a danger of data overflow if the answer log exceeds
+    # 1 MB. Given 1000-5000 answers, each answer must be at most
+    # 200-1000 bytes in size. We will address this later if it
+    # happens regularly. At the moment, a ValidationError is raised if
+    # an answer exceeds the maximum size.
+    MAX_BYTES_PER_ANSWER_VALUE = 500
+
+    # Prefix of strings that are cropped because they are too long.
+    CROPPED_PREFIX_STRING = 'CROPPED: '
+    # Answer value that is stored if non-string answer is too big
+    PLACEHOLDER_FOR_TOO_LARGE_NONSTRING = 'TOO LARGE NONSTRING'
+
+    # check type is dict
+    if not isinstance(answer_dict, dict):
+        raise utils.ValidationError(
+            'Expected answer_dict to be a dict, received %s' %
+            answer_dict)
+
+    # check keys
+    required_keys = set(REQUIRED_ANSWER_DICT_KEYS)
+    actual_keys = set(answer_dict.keys())
+    if not required_keys.issubset(actual_keys):
+        missing_keys = required_keys.difference(actual_keys)
+        raise utils.ValidationError(
+            'answer_dict is missing required keys %s' % missing_keys)
+
+    # check entries of answer_dict
+    if (sys.getsizeof(answer_dict['answer_value']) >
+            MAX_BYTES_PER_ANSWER_VALUE):
+
+        if type(answer_dict['answer_value']) == str:
+            logging.warning(
+                'answer_value is too big to be stored: %s ...' %
+            str(answer_dict['answer_value'][:MAX_BYTES_PER_ANSWER_VALUE]))
+            answer_dict['answer_value'] = (
+            '%s%s ...' %
```

Comment:
Fixed the indentation (and also for the warning statement before this statement).